src/preprocessing/passes/foreign_theory_rewrite.cpp

```diff
@@ -29,7 +29,72 @@ ForeignTheoryRewrite::ForeignTheoryRewrite(PreprocessingPassContext* preprocCont
     : PreprocessingPass(preprocContext, "foreign-theory-rewrite"),
       d_cache(preprocContext->getUserContext()){};
 
-Node ForeignTheoryRewrite::simplify(Node n) { assert(false); }
+Node ForeignTheoryRewrite::simplify(Node n)
+{
+  // first make sure the node is rewritten
+  n = Rewriter::rewrite(n);
+  // traverse the node
+  for (TNode current :
+       NodeDfsIterable(n, VisitOrder::POSTORDER, [this](TNode tn) {
+         return d_cache.find(tn) != d_cache.end();
+       }))
+  {
+    // the node is rewritten, and so GT, LT, LEQ
+    // should be eliminated
+    Assert(current.getKind() != kind::GT);
+    Assert(current.getKind() != kind::LT);
+    Assert(current.getKind() != kind::LEQ);
+
+    // for GEQ, we check whether the node can be simplified
+    if (current.getKind() == kind::GEQ)
```

Comment:
Put this in its own method.  Also, you should not stop at GEQ, instead traverse it.  The rewrite should be applied to fixed point, you should also traverse the results of the rewritten term.

Reply:
Thanks! fixed here: b5f7f6c